include/ozo/type_traits.h

```diff
@@ -403,6 +403,10 @@ constexpr auto size_of(const T& v) noexcept -> typename std::enable_if<
     return std::size(v) ? std::size(v) * size_of(*std::begin(v)) : 0;
 }
 
+struct name_oid {
+    std::string value;
+};
```

Comment:
May be [BOOST_STRONG_TYPEDEF](https://www.boost.org/doc/libs/1_67_0/libs/serialization/doc/strong_typedef.html)? Or in case if the mechanism is not convenient for us make something similar?

E.g.

  In a new file with name `include/ozo/pg_types.h`

```cpp

#include <boost/serialization/strong_typedef.hpp>

namespace ozo::pg {
    BOOST_STRONG_TYPEDEF(std::string, name)
} // namespace ozo::pg

```

Or maybe make extend a macro like this:

```cpp

#define OZO_STRONG_TYPEDEF(BASE, NEW)
    BOOST_STRONG_TYPEDEF(BASE, NEW)
    BASE ozo_get_underlying_type(NEW&&);

```

So, in this case you can reuse send_impl and recv_impl for the type strong defined.

Or just copy and paste Boost solution and extend it with necessary methods and types.

Reply:
Done